Source/WTF/wtf/CheckedRef.h

```diff
@@ -323,14 +336,26 @@ template <typename StorageType, typename PtrCounterType> class CanMakeCheckedPtr
                 WTFLogAlways("%s", stackTrace->toString().utf8().data());
         }
 #endif
-        // If you hit this assertion, you can turn on CHECKED_POINTER_DEBUG to help identify
-        // which CheckedPtr / CheckedRef outlived the object.
-        RELEASE_ASSERT(!m_count);
     }
 
     PtrCounterType ptrCount() const { return m_count; }
     void incrementPtrCount() const { ++m_count; }
-    void decrementPtrCount() const { ASSERT(m_count); --m_count; }
+    void decrementPtrCount() const
+    {
+        // In normal execution, a CheckedPtr always points to an object with a non-zero ptrCount().
+        // When it detects a dangling pointer, WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR scribbles an object with zeroes and then leaks it.
+        // When we check uncheckedPtrCount() here, we're checking for a scribbled object.
+        RELEASE_ASSERT(uncheckedPtrCount());
+        --m_count;
+    }
+
+    ALWAYS_INLINE PtrCounterType uncheckedPtrCount() const
```

Comment:
"Unchecked" means that the single-threaded CanMakeCheckedPtr will skip its "same thread" assertion. We need this because we call uncheckedPtrCount() after the destructor has run, when SingleThreadIntegralWrapper::m_thread is null.

Reply:
I see. Perhaps clarification in the commit message or a comment will help clarify.